internal/pkg/manifest/env.go

```diff
@@ -422,17 +425,18 @@ func (cfg *EnvironmentHTTPConfig) loadLBConfig(env *config.CustomizeEnv) {
 	if env.ImportVPC != nil && len(env.ImportVPC.PublicSubnetIDs) == 0 {
 		cfg.Private.InternalALBSubnets = env.InternalALBSubnets
 		cfg.Private.Certificates = env.ImportCertARNs
-		cfg.Private.SecurityGroupsConfig.Ingress.VPCIngress = aws.Bool(env.EnableInternalALBVPCIngress)
+		cfg.Private.Ingress.VPCIngress = env.EnableInternalALBVPCIngress
 		return
 	}
 	cfg.Public.Certificates = env.ImportCertARNs
 }
 
 // PublicHTTPConfig represents the configuration settings for an environment public ALB.
 type PublicHTTPConfig struct {
-	SecurityGroupConfig ALBSecurityGroupsConfig `yaml:"security_groups,omitempty"`
-	Certificates        []string                `yaml:"certificates,omitempty"`
-	ELBAccessLogs       ELBAccessLogsArgsOrBool `yaml:"access_logs,omitempty"`
+	SecurityGroupConfig DeprecatedALBSecurityGroupsConfig `yaml:"security_groups,omitempty"` // Deprecated. This configuration is now available inside Ingress field.
+	Certificates        []string                          `yaml:"certificates,omitempty"`
+	ELBAccessLogs       ELBAccessLogsArgsOrBool           `yaml:"access_logs,omitempty"`
+	Ingress             RestrictiveIngress
```

Comment:
I am confused 😝 is this field unmarshaled properly used?
If I look at the [proposed manifest schema](https://github.com/aws/copilot-cli/pull/4068#issuecomment-1275080333), I would have expected:
```
Ingress RestrictiveIngress  `yaml:"ingress"`
```
Am I missing something?

Reply:
Yes I got confused here with the multiple fields and thought that we are going to handle this ingress field internally through `security_groups` itself. But I am changing this one.